### PR TITLE
chore: migrate to GH Actions

### PR DESCRIPTION
**Pull request by jtmalinowski**
# Description

Replaces Circle CI with Github Actions.

Why? Circle CI is paid and only runs for users who are in the signalfx org. If we ever want to accept external contributions, we need to be able to run CI on them, and Circle is not suitable for that.

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

- [x] Tested manually
- [x] Added automated tests

# Checklist:

- [x] Changelogs have been updated
- [x] Unit tests have been added/updated
- [x] Documentation has been updated

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    branches:
+      - main
+permissions: read-all
+
+jobs:
+  unit-tests:
+    runs-on: ubuntu-latest
+    permissions: read-all
+    strategy:
+      fail-fast: false
```
Comment (owais) [inline .github/workflows/ci.yml:14]: Does this apply to each step for a nodejs version or the workflow as a whole? Makes sense to keep 12 running if 10 fails but do we really want to run compile if lint or test failed?

**Review by owais — APPROVED** [review]
LGTM. Suggest to allow to fail fast depending on what it refers to.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (Rauno56) [reply .github/workflows/ci.yml:14]: > You can allow specific jobs in a job matrix to fail without failing the workflow run. For example, if you wanted to only allow an experimental job with node set to 15 to fail without failing the workflow run.
From: https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstrategyfail-fast

So it's about different node versions, not steps, if I understand correctly.

**Review by Rauno56 — APPROVED** [review]
If I'm right on Owais's comment then LGTM.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (jtmalinowski) [reply .github/workflows/ci.yml:14]: Yeah it's about node versions, I agree in principle that a linting error could fail the whole matrix immediately and unit test in a specific Node.js versions shouldn't stop the matrix, but I don't think there's an easy way to do that.

**Pull request merged**